src/schema/posts.ts

```diff
@@ -1749,6 +1769,57 @@ export const resolvers: IResolvers<unknown, BaseContext> = traceResolvers<
         true,
       );
     },
+    fetchSmartTitle: async (
+      _,
+      { id }: { id: string },
+      ctx: Context,
+    ): Promise<GQLPostSmartTitle> => {
+      const post: Pick<Post, 'title' | 'contentMeta'> = await queryReadReplica(
+        ctx.con,
+        ({ queryRunner }) =>
+          queryRunner.manager.getRepository(Post).findOneOrFail({
+            where: { id },
+            select: ['title', 'contentMeta'],
+          }),
+      );
+      const isPlus = await isUserPlusMember(ctx.con, ctx.userId);
+      if (!isPlus) {
+        const hasUsedFreeTrial = await ctx.con
+          .getRepository(UserAction)
+          .findOneBy({
+            userId: ctx.userId,
+            type: UserActionType.FetchedSmartTitle,
+          });
+
+        if (hasUsedFreeTrial) {
+          throw new ForbiddenError(
+            'Free trial has been used! Please upgrade to Plus to continue using this feature.',
+          );
+        } else {
```

Comment:
no need to else since throw will kill the flow either way like early return

Reply:
👍